.github/workflows/publish-website.yml

```diff
@@ -0,0 +1,58 @@
+name: Deploy Website
+
+permissions:
+  contents: write
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - .github/**
+      - website/**
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/**
+      - website/**
+
```

Comment:
Also here, we have to build the website on every push, as a goal is to use the data from the ports to provide information about available features on the website (like what ports are there, what features do they have, ...)

Reply:
Which also means, we should collapse `WS_TARGET_ZIG_VERSION` and `MZ_TARGET_ZIG_VERSION` into a single version, as otherwise the two can easily get incompatible.

Also making them a secret prevents contributors to bump zig version via CI and requires "complex" interaction from a repository maintainer

Reply:
So I presume something like the std lib docs that zig has?

For collapsing them I've changed it to just `TARGET_ZIG_VERSION` for both WS and MZ.

For secrets vs variables. I think you still need a repository maintainer to update them do you not? According to GitHub (https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) you need to be an admin of the repo in order to set it up.

Reply:
> So I presume something like the std lib docs that zig has?

Yes, for example that. But also basically fetching the MicroZig package in the website and read all port information (so: which ports do we have, what chips and boards do they support, ...)